platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java

```diff
@@ -368,6 +378,12 @@ enum OverlappingOutputHandling {
      */
     default void validate(WorkValidationContext validationContext) {}
 
+    default void checkDependencies(WorkValidationContext validationContext) {}
```

Comment:
💭 I wonder how we should call this. This now only checks if the outputs of the work (task) overlap with any already known inputs.

Reply:
Good point. I've improved the naming and the structure of `MissingTaskDependencyDetector`. I've also added a bunch of Javadoc.